From f4b571edecc36969df4a78e6cf7aea50235e8837 Mon Sep 17 00:00:00 2001
From: alexzk1 <alexzkhr@gmail.com>
Date: Thu, 5 Jul 2018 01:09:47 +0300
Subject: [PATCH] change to use isFast

---
 src/llvm/BitcodeReader.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/llvm/BitcodeReader.cpp b/src/llvm/BitcodeReader.cpp
index 226bc47..c9cd6d1 100644
--- a/src/llvm/BitcodeReader.cpp
+++ b/src/llvm/BitcodeReader.cpp
@@ -617,7 +617,11 @@ static intermediate::InstructionDecorations toInstructionDecorations(const llvm:
         deco = add_flag(deco, intermediate::InstructionDecorations::NO_INF);
     if(llvm::isa<llvm::FPMathOperator>(&inst) && inst.hasAllowReciprocal())
         deco = add_flag(deco, intermediate::InstructionDecorations::ALLOW_RECIP);
+#if LLVM_LIBRARY_VERSION >= 60  
+    if(llvm::isa<llvm::FPMathOperator>(&inst) && inst.isFast())
+#else
     if(llvm::isa<llvm::FPMathOperator>(&inst) && inst.hasUnsafeAlgebra())
+#endif        
         deco = add_flag(deco, intermediate::InstructionDecorations::FAST_MATH);
     return deco;
 }